### ISSUE TITLE
Allow %editing ipy files with -r

### ISSUE THREAD

**Issue by memeplex** [ipython/ipython#9100]
I think `%edit -r /tmp/something.ipy` could be useful in a way parallel to `%edit /tmp/something.py`.

**Comment by memeplex** [comment]
My main use case is to edit magic cells terminal ipython.

Some alternatives are:
1. Special parameter to `%edit` cells. Say `%edit -c sh`.
2. Some hot key to open an external editor (say vim) while editing, with the current ongoing edition (that would be very useful in a lot of cases! As in bash `fc` which could be bound to some key).

**Comment by takluyver** [comment]
Does editing `.ipy` files not already work?

I'm working on making IPython use prompt_toolkit internally, and I think that has some shortcuts to open the current buffer in an external editor.

**Comment by memeplex** [comment]
%edit doesn't recognize the .ipy extension, just .py which makes sense considering it will exec the contents. For the -r case I think it's sensible to recognize .ipy as a file extension.

Just out of curiosity, there is ptipython, which is the advantage of using pt internally? And will you still support readline?

**Comment by takluyver** [comment]
Yes, it probably makes sense to be able to do `%edit foo.ipy`. There's no need for the `-r` flag, that's just to get the raw format when you're editing from history (e.g. `%edit -r 2-7`).

I know of ptipython, and we're basically going to fold the technology into ipython itself and make ptipython obsolete. prompt_toolkit is for terminal interaction, so it's not relevant to jupyter clients like the notebook or the Qt console. I plan to work it into terminal IPython first, and later jupyter_console. I expect that we will eventually drop support for readline, because it's an awkward dependency (we have [a docs section](http://ipython.readthedocs.org/en/stable/install/install.html#dependencies) specific to installing it) and prompt_toolkit is more capable.

**Comment by memeplex** [comment]
A mapping from file extensions to cell magics could be kept and those file types could be evaluated using their magics. This would provide with powerfull cell editing in the console. The raw editing solution I proposed above is somewhat lacking because it implies to type the cell heading and to lose editor features contextual to the file type, because of the ipy extension. What do you think? I could code this and make a pr if you agree.

I'm glad you're adding prompt_toolkit support and that eventually it will make its way into the jupyter console, but please never drop good old readline support as the julia guys did.
